entry.go

```diff
@@ -29,16 +31,23 @@ type Entry struct {
 
 	// Message passed to Debug, Info, Warn, Error, Fatal or Panic
 	Message string
+
+	Depth int
 }
 
-func NewEntry(logger *Logger) *Entry {
+func NewEntry(logger *Logger, d int) *Entry {
 	return &Entry{
 		Logger: logger,
 		// Default is three fields, give a little extra room
-		Data: make(Fields, 5),
+		Data:  make(Fields, 5),
+		Depth: d,
 	}
 }
 
+func (entry *Entry) deepen() {
+	entry.Depth = entry.Depth + 1
```

Comment:
As far as I know, when you call to log, a new entry is created. So no data race is possible.